server/src/MessageHandler.ts

```diff
@@ -0,0 +1,144 @@
+import {
+  TextDocumentPositionParams,
+  CancellationToken,
+  TextDocuments,
+  DocumentFormattingParams,
+  TextEdit,
+  Range,
+  Location,
+  DeclarationParams,
+  CompletionParams,
+  CompletionItem,
+  CompletionItemKind,
+} from 'vscode-languageserver'
+import * as util from './util'
+import { fullDocumentRange } from './provider'
+import { getDMMF } from '@prisma/sdk'
+import { TextDocument, Position } from 'vscode-languageserver-textdocument'
+import format from './format'
+
+export class MessageHandler {
+  constructor() {}
+
+  getWordAtPosition(document: TextDocument, position: Position): string {
+    const currentLine = document.getText({
+      start: { line: position.line, character: 0 },
+      end: { line: position.line, character: 9999 },
+    })
+    // search for the word's beginning and end
+    const beginning = currentLine
+      .slice(0, position.character + 1)
+      .search(/\S+$/)
+    const end = currentLine.slice(position.character).search(/\W/)
+    if (end < 0) {
+      return ''
+    }
+    return currentLine.slice(beginning, end + position.character)
+  }
+
+  async handleDefinitionRequest(
+    documents: TextDocuments<TextDocument>,
+    params: DeclarationParams,
+    _token?: CancellationToken,
+  ): Promise<Location> {
+    // TODO: Replace bad workaround as soon as ASTNode is available
+
+    const textDocument = params.textDocument
+    const position = params.position
+
+    const document = documents.get(textDocument.uri)
+
+    if (!document) {
+      return new Promise((resolve) => resolve())
+    }
+
+    const documentText = document.getText()
+
+    const word = this.getWordAtPosition(document, position)
+    if (word === '') {
+      return new Promise((resolve) => resolve())
+    }
+
+    // parse schem file to datamodel meta format (DMMF)
+    const dmmf = await getDMMF({ datamodel: documentText })
+
+    const modelName = dmmf.datamodel.models
+      .map((model) => model.name)
+      ?.find((name) => name === word)
+
+    // selected word is not a model type
+
+    if (!modelName) {
+      return new Promise((resolve) => resolve())
```

Comment:
```suggestion
      return Promise.resolve()
```

Reply:
this throws an error, bc type 'void' is not assignable to to type 'Location'